### ISSUE TITLE
Causes site to crash when using PHP 7.3

### ISSUE THREAD

**Issue by desmondhoward** [wpengine/atlas-content-modeler#393]
Site using PHP 7.3 crashes once plugin is activated with `Fatal error: Uncaught ArgumentCountError: array_merge() expects at least 1 parameter, 0 given`.

### To reproduce
1. Set PHP version to 7.3
2. Upload & activate plugin

### Expected behavior
Plugin doesn't crash site. Using PHP 7.4 the site works as expected.

### Screenshots
<img width="708" alt="Screen Shot 2022-01-07 at 9 57 54 AM" src="https://user-images.githubusercontent.com/96273496/148562029-dc79037c-4fa7-43b1-988b-eff71ee28cec.png">

### Version information
- Plugin version: 0.12.0
- WordPress version: 5.8.3
- Operating system: MacOS locally, but also happened in production
- Browser: Google Chrome

### Additional context
- [Line causing the error](https://github.com/wpengine/atlas-content-modeler/blob/main/includes/content-registration/custom-post-types-registration.php#L758)
- [PHP Backward Incompatible docs for "Throw on passing too few function arguments"](https://www.php.net/manual/en/migration71.incompatible.php)

**Comment by mindctrl** [comment]
@desmondhoward thanks for the report. I was not able to reproduce this on PHP 7.3 using the instructions you listed. Looking at the line of code causing the error, it seems like it would happen when there are no fields on a model. I see some improvements we can make, but I'm curious how exactly you were able to trigger this. Do you have other plugins active on the site?

**Comment by desmondhoward** [comment]
there are other plugins running on the site. to trigger this I just installed/activated the plugin on a site that has no models/fields.

after some more digging, I think I've found the issue: 
**[starting in PHP 7.4](https://www.php.net/manual/en/function.array-merge.php#refsect1-function.array-merge-changelog), `array_merge()` "can now be called without any parameter"**. before, at least 1 was required.

so I think since PHP 7.1+ throws when there are too few arguments and `array_merge()` expects at least 1 in < PHP 7.4, this code is expected to fail when running PHP 7.1 - 7.3 with no models.
